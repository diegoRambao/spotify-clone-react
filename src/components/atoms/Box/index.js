import styled from 'styled-components'

const Container = styled.div`
    margin-top: ${props => props.mt};
    padding-top: ${props => props.pt};
    margin-right: ${props => props.mr};
    margin-left: ${props => props.ml};
`

export function Box ({ children, mt, mr, ml, pt, className }) {
  return (
    <Container
      className={className}
      ml={ml}
      mr={mr}
      mt={mt}
      pt={pt}
    >{children}
    </Container>
  )
}
