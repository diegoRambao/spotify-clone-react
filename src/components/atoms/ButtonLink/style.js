import styled from 'styled-components'
import { colors } from '../../../consts/styles'

export const ButtonContainer = styled.a`
    background-color: ${colors.primary};
    color: ${colors.white};
    text-decoration: none;
    border-radius: 25px;
    padding: 14px 28px;
    font-size: 18px;
    font-weight: 600;
    display: block;
    text-align: center;
`
