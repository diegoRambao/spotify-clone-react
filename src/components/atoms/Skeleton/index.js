import styled, { keyframes } from 'styled-components'

const skeletonFrames = keyframes`
    100% {
      transform: translateX(100%);
    }
`

const Container = styled.span`
  display: inline-block;
  height: 1em;
  position: relative;
  overflow: hidden;
  background-color: rgba(255, 255, 255, 0.13);
  width: ${props => props.width || '100%'};
  height: ${props => props.height || '100%'};
  border-radius: 0;
  border-radius: ${props => props.variant === 'circle' && '50%'};
  border-radius: ${props => props.variant === 'rect' && '2px'};

  &::after {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    transform: translateX(-100%);
    background: linear-gradient(
90deg
, transparent, rgba(255, 255, 255, 0.08), transparent);
    animation: ${skeletonFrames} 2s infinite;
    content: '';
  }
`

export function Skeleton ({ variant, width, height }) {
  return <Container width={width} height={height} variant={variant} />
}
