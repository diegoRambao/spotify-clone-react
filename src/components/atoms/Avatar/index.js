import { AvatarImage, AvatarContainer } from './style'

export function Avatar ({ children, src, alt }) {
  return (
    <AvatarContainer>
      {children && children}
      {!children && <AvatarImage
        id='img'
        src={src}
        alt={alt}
                    />}

    </AvatarContainer>
  )
}
