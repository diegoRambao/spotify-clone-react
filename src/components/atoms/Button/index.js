import { ButtonContainer } from './style'

export function Button ({ onClick, children }) {
  return (
    <ButtonContainer onClick={onClick}>
      {children}
    </ButtonContainer>
  )
}
