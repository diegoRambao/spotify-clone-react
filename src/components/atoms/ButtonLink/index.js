import { ButtonContainer } from './style'

export function ButtonLink ({ href, children }) {
  return (
    <ButtonContainer href={href}>
      {children}
    </ButtonContainer>
  )
}
