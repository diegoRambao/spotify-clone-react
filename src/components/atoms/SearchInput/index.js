import { useRef } from 'react'
import { InputRoot, TextField, Icon } from './style'

export function SearchInput ({ onSubmit }) {
  const textFieldEl = useRef()

  const handleEnter = (event) => {
    if (event.key === 'Enter') {
      const { value } = textFieldEl.current
      onSubmit(value)
    }
  }

  return (
    <InputRoot>
      <Icon />
      <TextField
        ref={textFieldEl}
        onKeyDown={handleEnter}
        type='text'
        name='search'
        spellCheck='false'
        placeholder='Buscar canciones...'
        autoFocus
      />
    </InputRoot>
  )
}
