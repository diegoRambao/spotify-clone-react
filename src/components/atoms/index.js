import { Avatar } from './Avatar'
import { Box } from './Box'
import { SearchInput } from './SearchInput'
import { Skeleton } from './Skeleton'
import { Spiner } from './Spiner'

export {
  Avatar,
  Box,
  SearchInput,
  Skeleton,
  Spiner
}
