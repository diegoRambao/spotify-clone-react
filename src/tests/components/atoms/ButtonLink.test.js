import { shallow, mount } from 'enzyme'
import { ButtonLink } from 'components/atoms/ButtonLink'

describe('<ButtonLink />', () => {
  const text = 'Text example'
  const url = 'https://www.google.com'
  const wrapper = mount(<ButtonLink href={url}>{text}</ButtonLink>)

  test('render content', () => {
    const wrapper = shallow(<ButtonLink>{text}</ButtonLink>)
    expect(wrapper).toMatchSnapshot()
  })

  test('render children', () => {
    expect(wrapper.text()).toBe(text)
  })

  test('to have attribute href', () => {
    expect(wrapper.prop('href')).toBe(url)
  })
})
