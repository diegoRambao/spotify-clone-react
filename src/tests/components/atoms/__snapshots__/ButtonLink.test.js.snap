// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonLink /> render content 1`] = `
<styled.a>
  Text example
</styled.a>
`;
